packages/celotool/src/e2e-tests/transfer_tests.ts

```diff
@@ -829,4 +867,78 @@ describe('Transfer tests', function(this: any) {
+            await freeze('http://localhost:8545', validatorAddress, CeloContract.GoldToken)
+          })
+
+          describe('check if frozen', () => {
+            it('should be frozen', async () => {
+              const goldTokenAddress = await kit.registry.addressFor(CeloContract.GoldToken)
+              const freezer = await kit._web3Contracts.getFreezer()
+              const isFrozen = await freezer.methods.isFrozen(goldTokenAddress).call()
+              assert(isFrozen)
+            })
+          })
+          describe('when neither sender nor receiver is whitelisted', () => {
+            before('ensure neither sender nor receiver is whitelisted', async () => {
+              await setAddressWhitelist('http://localhost:8545', validatorAddress, [])
+            })
+            it('should not add the transaction to the pool', async () => {
+              const gas = INTRINSIC_TX_GAS_COST + ADDITIONAL_INTRINSIC_TX_GAS_COST
+              const feeCurrency = await kit.registry.addressFor(CeloContract.StableToken)
+              try {
+                const res = await transferCeloGold(FromAddress, ToAddress, TransferAmount, {
+                  gas,
+                  feeCurrency,
+                })
+                await res.waitReceipt()
+                assert.fail('no error was thrown')
+              } catch (error) {
+                assert.include(error.toString(), 'Returned error: transfers are currently frozen')
+              }
+            })
+          })
+          describe('when receiver is whitelisted', () => {
+            it('should transfer succesfully', async () => {
+              const whitelistedAddress = await kit.registry.addressFor(CeloContract.LockedGold)
+              testTransferToken({
+                expectedGas: INTRINSIC_TX_GAS_COST + ADDITIONAL_INTRINSIC_TX_GAS_COST,
+                transferToken: CeloContract.GoldToken,
+                feeToken: CeloContract.GoldToken,
+                toAddress: whitelistedAddress,
+              })
+            })
+          })
+          describe('when sender is whitelisted', () => {
+            before('add sender to transfer whitelist', async () => {
+              await whitelistAddress('http://localhost:8545', validatorAddress, FromAddress)
+            })
+            it('should transfer succesfully', async () => {
+              testTransferToken({
+                expectedGas: INTRINSIC_TX_GAS_COST + ADDITIONAL_INTRINSIC_TX_GAS_COST,
+                transferToken: CeloContract.GoldToken,
+                feeToken: CeloContract.GoldToken,
+              })
+            })
+            after('remove sender from transfer whitelist', async () => {
+              await setAddressWhitelist('http://localhost:8545', validatorAddress, [])
+            })
+          })
+
+          after('unfreeze gold transfers', async () => {
+            await unfreeze('http://localhost:8545', validatorAddress, CeloContract.GoldToken)
```

Comment:
Similarly, can we add a check to ensure that unfreezing the contract allows transfers?

Reply:
This is kinda already checked in "Normal Transfer >" but for the sake of testing the unfreeze helper I've added a test block that calls unfreeze in the before hook. The top-level before hook calls freeze(), so this should check that unfreeze does what it's supposed to.